code/addons/test/src/postinstall.ts

```diff
@@ -51,57 +55,158 @@ export default async function postInstall(options: PostinstallOptions) {
       ? info.rendererPackageName
       : null;
 
-  if (!annotationsImport) {
-    logger.info('The Vitest addon cannot yet be used with: ' + info.frameworkPackageName);
+  const prerequisiteCheck = async () => {
+    const reasons = [];
+
+    if (
+      info.frameworkPackageName !== '@storybook/nextjs' &&
+      info.builderPackageName !== '@storybook/builder-vite'
+    ) {
+      reasons.push(
+        'The Vitest addon can only be used with a Vite-based Storybook framework or Next.js.'
+      );
+    }
+
+    if (!annotationsImport) {
+      reasons.push(dedent`
+        The Vitest addon cannot yet be used with ${colors.pink.bold(info.frameworkPackageName)}
+      `);
+    }
+
+    const vitestVersion = await packageManager.getInstalledVersion('vitest');
+    if (vitestVersion && !satisfies(vitestVersion, '>=2.0.0')) {
+      reasons.push(`
+        The Vitest addon requires Vitest 2.0.0 or later.
+        Please update your ${colors.pink.bold('vitest')} dependency and try again.
+      `);
+    } else {
+      const depRanges = await packageManager.getAllDependencies();
+      const checkDependencies = [...dependencies, ...optionalDependencies];
+      const latestVersions = await packageManager.getVersions(...checkDependencies);
+      const latestPackages = checkDependencies.map(
+        (pkg, index) => [pkg, latestVersions[index].replace('^', '')] as const
+      );
+
+      let hasInconsistentPackageVersions = false;
+      for (const [pkg, latestVersion] of latestPackages) {
+        if (depRanges[pkg]) {
+          if (!dependencies.includes(pkg)) {
+            // Add found optional dependency so it will be updated to the latest version
+            dependencies.push(pkg);
+          }
+
+          if (!satisfies(latestVersion, depRanges[pkg])) {
+            hasInconsistentPackageVersions = true;
+            reasons.push(dedent`
+              The package ${colors.pink.bold(pkg)} is already installed and cannot be updated to ${colors.pink.bold(latestVersion)} because it would not satisfy "${colors.pink.bold(depRanges[pkg])}".
```

Comment:
There's no prior indication here about _why_ you want to update to the latest anyway. As a user, my reaction here is "well, then don't update!?!"?

Reply:
We are removing this check altogether